crates/qt-build-utils/src/lib.rs

```diff
@@ -224,515 +136,62 @@ pub struct QmlModuleRegistrationFiles {
-                        error: Box::new(e),
-                    })
-                }
-            }
-        }
-
-        // Fedora 36 renames Qt5's qmake to qmake-qt5
-        let candidate_executable_names = ["qmake6", "qmake-qt5", "qmake"];
-        for (index, executable_name) in candidate_executable_names.iter().enumerate() {
-            match verify_candidate(executable_name) {
-                Ok((executable_name, version)) => {
-                    return Ok(Self {
-                        qmake_executable: executable_name.to_string(),
-                        moc_executable: None,
-                        qmltyperegistrar_executable: None,
-                        qmlcachegen_executable: None,
-                        rcc_executable: None,
-                        version,
-                        qt_modules,
-                    });
-                }
-                // If QT_VERSION_MAJOR is specified, it is expected that one of the versioned
-                // executable names will not match, so the unversioned `qmake` needs to be
-                // attempted last and QtVersionMajorDoesNotMatch should only be returned if
-                // none of the candidate executable names match.
-                Err(QtBuildError::QtVersionMajorDoesNotMatch {
-                    qmake_version,
-                    qt_version_major,
-                }) => {
-                    if index == candidate_executable_names.len() - 1 {
-                        return Err(QtBuildError::QtVersionMajorDoesNotMatch {
-                            qmake_version,
-                            qt_version_major,
-                        });
-                    }
-                    eprintln!("Candidate qmake executable `{executable_name}` is for Qt{qmake_version} but QT_VERSION_MAJOR environment variable specified as {qt_version_major}. Trying next candidate executable name `{}`...", candidate_executable_names[index + 1]);
-                    continue;
-                }
-                Err(QtBuildError::QtMissing) => continue,
-                Err(e) => return Err(e),
-            }
-        }
-
-        Err(QtBuildError::QtMissing)
-    }
-
-    /// Get the output of running `qmake -query var_name`
-    pub fn qmake_query(&self, var_name: &str) -> String {
-        std::str::from_utf8(
-            &Command::new(&self.qmake_executable)
-                .args(["-query", var_name])
-                .output()
-                .unwrap()
-                .stdout,
-        )
-        .unwrap()
-        .trim()
-        .to_string()
+    /// Create a [QtBuild] using the default [QtInstallation] (currently uses [QtInstallationQMake])
+    /// and specify which Qt modules you are linking, ommitting the `Qt` prefix (`"Core"`
+    /// rather than `"QtCore"`).
+    //
+    // TODO: is there a better name for this method or a sane way to create a default QtInstallation?
+    pub fn new_with_default_installation(qt_modules: Vec<String>) -> anyhow::Result<Self> {
```

Comment:
This should be `new` and the other one should be `with_installation` as "automatically doing the right thing" should be the behavior of `new`.